service/src/test/java/bio/terra/catalog/service/dataset/DatasetDaoTest.java

```diff
@@ -0,0 +1,68 @@
+package bio.terra.catalog.service.dataset;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.catalog.app.App;
+import bio.terra.catalog.service.dataset.exception.InvalidDatasetException;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ContextConfiguration(classes = App.class)
+class DatasetDaoTest {
+
+  @Autowired private DatasetDao datasetDao;
+  @Autowired private NamedParameterJdbcTemplate jdbcTemplate;
+
+  private enum storageSystem {
+    TERRA_WORKSPACE,
+    TERRA_DATA_REPO
+  };
+
+  @Test
+  void testCreateDeleteDataset() throws Exception {
+    String datasetId = UUID.randomUUID().toString();
+    Dataset dataset =
+        new Dataset()
+            .datasetId(datasetId)
+            .storageSystem("DATA_REPO")
+            .metadata("{\"sampleId\": \"12345\", \"species\": [\"mouse\", \"human\"]}");
+    Dataset duplicateDatasetId =
+        new Dataset()
+            .datasetId(datasetId)
+            .storageSystem(String.valueOf(storageSystem.TERRA_WORKSPACE));
+    try {
+      datasetDao.create(dataset);
+      datasetDao.create(duplicateDatasetId);
+      int datasetCount = datasetDao.enumerate().size();
+      assertEquals(datasetCount, 2);
+    } finally {
+      datasetDao.enumerate().forEach(d -> datasetDao.delete(d.getId()));
```

Comment:
If the IDs are random, do we need to delete the data created? The unit tests will only be run against local databases (that is, never against dev or other persistent environments). I think it would be better to not have each test clean up after itself, in order to keep the test code simple.

Reply:
We could move the delete to a separate cleanup method to simplify the tests, but I don't feel strongly either way